src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,78 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class WorkWithFile {
+    private StringBuilder dataToWrite = new StringBuilder();
 
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> dataFromFile = readDataFromFile(fromFileName);
+        String finalArrayOfData = createReport(dataFromFile, new String[2],0,0);
+
+        writeDataToFile(toFileName, finalArrayOfData);
+    }
+
+    public List<String> readDataFromFile(String fromFileName) {
+        List<String> res = new ArrayList<>();
+        String line;
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = reader.readLine()) != null) {
+                res.addAll(Arrays.asList(line.split(",")));
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file: " + fromFileName, e);
+        }
+        return res;
+    }
+
+    public void writeDataToFile(String fileName, String dataToWrite) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(fileName))) {
+            writer.write(dataToWrite);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't find a file: " + fileName, e);
+        }
+    }
 
+    public String createReport(List<String> data, String[] result,
+                               int sumOfTheValuesOfTheFirstElements,
+                               int sumOfTheValuesOfTheSecondElements) {
```

Comment:
```suggestion
    public String createReport(List<String> lines) {
```
 you don't need to pass all these parameters
create two variables for a total sum of supply and a total sum of buy
go through all lines using for loop and split each line and get all needed data to decide to which variable you need to add a number in a line

Reply:
I don't understand this solution in what aspects is my approach worse?

Reply:
if you have questions ask in chats, mentors don't answer here

Reply:
Updated